clause_6_13_ModelTextures.adoc

```diff
@@ -579,3 +579,38 @@ The <Material> tag presented in section 6.5.3 is a high level mean of providing
 In short, the <Material> tag supports a polygon-based approach of sensor client devices such as FLIR, NVG, and RADAR. A Material texture is a texel-based approach supporting an implementation of such client devices with a much higher resolution.
 
 In the case of the Raster Material dataset (dataset code 005) applied onto the terrain, it is conceivable that multiple layers and mixtures of materials are required to represent the rich variety of materials found on the earth surface. However, for Models, a single material layer is probably adequate for the vast majority of man-made objects.
+
+===== Model Heat Map Texture
+
+A Heat Map texture describes the temperature radiated by the surface described by its base textrue (ex. engine). A heat map is stored as an 8-bit single channel texture (a grey-scale image) where texels are mapped to the range 0.0 (ambient temperature) to 1.0 (hot). This texture can be used to simulate IR sensor effects.
+
+image::images/Sensor-sidebyside.png[image,width=475,height=359]
+[#img_HeatMapSideBySide,reftext='{figure-caption} {counter:figure-num}']
+*_{figure-caption}{counter:figure-num}. Base with corresponding heap map texture_*
+
+
+image::images/Sensor-Rendering.png[image,width=475,height=359]
+[#img_RenderedHeatMapinasimulatedIRsensor,reftext='{figure-caption} {counter:figure-num}']
+*_{figure-caption}{counter:figure-num}. Rendered Heat Map in a simulated IR sensor_*
+
+===== Model Decal Texture
+
+A decal texture is overlayed on top of the base texture to provide extra detail (sticker, crack, etc.)
+
+===== Model Mask Texture
+
+A mask texture is used to control the transparancy of its base texture, independently of its mapping. The mask texture will multiply the transparency of the base texture. A mask texture is stored as an 8-bit single channel texture (a grey-scale image) where texels are mapped to the range 0.0 (0-transparent) to 1.0 (1-opaque). 
+
+Below is an example of a mask texture that can be used to feather the hard edges from a base texture.
+
+image::images/Mask.png[image,width=475,height=359]
+[#img_Mask,reftext='{figure-caption} {counter:figure-num}']
+*_{figure-caption}{counter:figure-num}. Mask Texture_*
+
+===== Model ORM Texture
+
+An ORM texture is used for Physically Based Rendering (PBR) effects. An ORM texture is a 3 channel packed texture defining the Ambient Occlusion (O), Roughness \(R) and Metalness (M) of the surface.
+
```

Comment:
From what (little) I understand of PBR, there are two models (some details here: https://www.a23d.co/blog/pbr-textures-metallic-vs-specular-workflow/):
- Albedo/Roughness/Metalness;
- and Diffuse/Specular/Glossiness.

An occlusion map is somewhat distinct from either model.
A few questions:

- With this "ORM" model, where does the Albedo map come from? Is it assumed that the old "diffuse" texture then represents albedo?
- Are there plans to support both models (in CDB 1.x or 2.0)? (we plan to support for both in our [E3D models spec](https://github.com/ecere/E3D-spec#detailed-description-of-all-block-types)). I think Unreal only supports the Roughness/Metalness model ( https://dev.epicgames.com/community/learning/tutorials/Yx3q/unreal-engine-physically-based-rendering-pbr-explained-in-depth ), but Unity supports both ( https://docs.unity3d.com/Manual/StandardShaderMetallicVsSpecular.html ).
- Would it make sense to separate out the occlusion map, as it is not directly related to the PBR properties?

Reply:
This particular texture approach differs substantially from glTF (as I understand it) since that approach has a separate occlusion texture (which can be used in any shading model not just metal-roughness) and the metal-roughness texture is its own 2-channel thing. Is there a reference to where this particular ORM approach came from and why it was chosen? Is this based on any existing standard?

As best I understand Unity, you can map any of the standard shader inputs to particular sampler channels so it's not too picky on that, but other things may barf if the texture layout isn't following the glTF approach since glTF is the main model interchange format that currently supports this sort of thing.

Unity and glTF also supports alternate specular-glossiness PBR model, where a separate specular color texture is needed along with a 1-channel glossiness input.

There are other more advanced PBR models even beyond this, but I do not think they are yet widely used.

Reply:
Discussed in the CDB SWG Meeting on 3/10/2023.  Several commenters agreed to attempt to resolve the perceived consensus during the SWG meeting that an update to the Pull Request could be created with three-channels; and that such an update would be backwards-compatible; and that such an update could accomodate Robert's original intent, as well as better support for Unity, as well as complete (or nearly-complete) 'lift' of the PBR approach in glTF.  Comments and inputs to resolve this conversation should be entered here in the PR please.  The vote has already passed and achieved quorum, but when we hold our next SWG meeting on 3/31/2023 (the closing date of this vote), we will ask the voters who voted before this note if there is any objection to the changes in the resolution of the issue/conversion here.

Reply:
> Albedo = base texture

The problem is that a proper Albedo texture used together with a Roughness/Metalness texture will not be the same as a classic Diffuse texture.
Therefore a CDB could not accommodate both PBR and legacy engines if an Albedo texture type distinct from the OpenFlight Diffuse texture does not exist.

I like the option to define both ORM as well as separate RM and O tetures.

Reply:
I've added RM and O dataset types to volume 1.

Reply:
I'm not sure I follow what you mean by "distinct from the OpenFlight Diffuse texture". In OpenFlight there is no diffuse texture, it is a base texture. In OpenFlight you can specify Ambient, Diffuse, Specular, Emissive properties in the material of the polygon, but these are scalar properties, not a texture.

Reply:
@RobertRicci That OpenFlight "base texture" is a "diffuse texture". The scalar Diffuse property is multiplied by the "base texture" color and gets used as the material's "diffuse" property in the [Blinn-Phong reflection model](https://en.wikipedia.org/wiki/Blinn%E2%80%93Phong_reflection_model).
It is different from e.g., the `baseColorTexture` in GLTF Metallic-Roughness PBR materials, which is an albedo texture.

For example, I believe the OpenFlight base texture may contain features like scratches which in the Albedo-Roughness-Metalness PBR model would be in the roughness texture instead.

Reply:
@RobertRicci 
This might explain better than I can:

https://www.a23d.co/blog/difference-between-albedo-and-diffuse-map/

Reply:
In OpenFlight, if you re intending to use multi-texturing techniques (ex. normal map) or ORM texture in order to have a realistic lighting realism, then you would simply use an albedo as your base texture. I still don't understand how you assume the base texture is a diffused type? Nothing prevents you from using one, but then this defies the purpose of using other additive techniques.

Reply:
So part of the issue here is that OpenFlight does not contain an occlusion texture in any of their material formats.  So their diffuse texture is albedo + occlusion, right?  

@RobertRicci - So how would you encode this in OpenFlight?  Are you planning on using the extended material palette (placing the albedo in the diffuse slot) or just using the per-polygon stack of textures (where CDB could have a base texture be albedo, with a layer for occlusion and another layer for roughness/metalness)?

Reply:
No, it is done using the conventional multi-texturing technique. You currently have the possibility to have 7 additional layers apart from the base. We are using he additional layers for light maps, micro, normal, etc. Here we would have an additional layer pointing to an ORM texture.

Reply:
@RobertRicci 

A "normal map" is orthogonal realism capability that is compatible with both a Blinn-Phong diffuse map and a PBR albedo map. The normal map changes the direction of the light vector.

There in an assumption that the "base texture" is a diffuse map in most pre-PBR legacy systems, so I am quite certain this applies to OpenFlight / CDB 1.x as well, but others like @ryanfranz may know better than myself.

Reply:
Suggested way to present albedo / diffuse compatibility rules for a new distinct 010 albedo map:


New Albedo texture type (010) for PBR

```
(base) 
Diffuse  (O)RM   Albedo   Legacy System                                          PBR System

   Y       N        N    Diffuse: Diffuse                                        Albedo: Diffuse Map (with deriving logic, if supported)
   Y       Y        N    Diffuse: Diffuse (assuming it is albedo, if supported)  Albedo: Diffuse Map (as-is)
   Y       Y        Y    Diffuse: Diffuse                                        Albedo: Albedo Map
   N       Y        Y    Diffuse: Albedo (with deriving logic, if supported)     Albedo: Albedo Map
   N       N        Y    Diffuse: Albedo (as-is)                                 Albedo: Albedo Map
```

Reply:
@jerstlouis, @ryanfranz, I've pushed new changes for vol-1 and vol-6, this is the addition of texture type 10 and added definition for diffuse in section 6.13.1 of vol-6. Do you agree we can add the above table in section 6.13.1 right after the line I added on the definition of diffuse texture?  I don't think this should go in Annex B (OpenFlight) as the is not really about how to encode, but more about how to interpret the meaning of the base texture.  Also any recommendation on how to add this table to aDoc?

Reply:
The easiest kluge I've found for creating a table is to use Google Docs to create the table, then using the Google Docs to ASCIIDOCTOR conversion tool.  I will take a shot at doing that with the table a little later this morning

Reply:
Here's the table converted to ASCIIDOC

|===
|(base) |(O)RM| Albedo| Legacy System  | PBR System

|Y|N|N|  Diffuse: Diffuse|Albedo: Diffuse Map (with deriving logic, if supported)
|Y|Y|N| Diffuse: Diffuse (assuming it is albedo, if supported)| Albedo: Diffuse Map (as-is)
|Y|Y|Y| Diffuse: Diffuse  |Albedo: Albedo Map
|N|Y|Y|Diffuse: Albedo (with deriving logic, if supported)|Albedo: Albedo Map
|N|N|Y| Diffuse: Albedo (as-is)     |Albedo: Albedo Map 
|===

Reply:
Thanks David. Added table to vol 6.